### PR TITLE
[#176082784] Refactor push notification logic

### PR DESCRIPTION
**Pull request by gunzip**
Removes orchestration and just call the notification hub API directly (the queue trigger function has 10 retries on its own by default) in order to speed up push notifications delivery.

Draft since I'm not sure this would be a better choice than using the orchestrator:
1) it looks like the notification / installation queue is emptied every day anyway, which means at worst a push takes 12h to being delivered
2) by default the queue trigger retries using short intervals (seconds), so if the notification hub service is unreachable push notifications have a high probabilty to fail
3) if we introduce the retry policy for the function, it is not clear from the documentation where this state is kept, so we don't know if it can scale properly

### CODE REVIEW

File: HandleNHNotificationCall/index.ts [hunk 1]
```diff
@@ -22,11 +24,21 @@ initTelemetryClient();
  */
 export async function index(
   context: Context,
-  notificationHubMessage: NotificationHubMessage
+  input: NotificationHubMessage
```
Comment (francescopersico) [inline HandleNHNotificationCall/index.ts:27]: I think the type must be `unknown`.

**Review by francescopersico — APPROVED** [review]

File: HandleNHNotificationCallActivity/handler.ts [hunk 2]
```diff
@@ -98,9 +101,22 @@ export const getCallNHServiceActivityHandler = (
...
+                  name: "api.messages.create.push",
+                  properties: {
+                    isSuccess: "true",
+                    messageId: message.payload.message_id
+                  },
+                  tagOverrides: { samplingEnabled: "false" }
+                })
+              )
+            );
         case DeleteKind.DeleteInstallation:
```
Comment (AleDore) [inline HandleNHNotificationCallActivity/handler.ts:120]: Why track only notify events?

File: HandleNHNotificationCallActivity/handler.ts [hunk 3]
```diff
@@ -98,9 +102,22 @@ export const getCallNHServiceActivityHandler = (
...
-            retryActivity(context, `${logPrefix}|ERROR=${toString(e)}`)
-          );
+          return notify(message.installationId, message.payload)
+            .mapLeft(e =>
+              retryActivity(context, `${logPrefix}|ERROR=${toString(e)}`)
+            )
+            .chainFirst(
+              taskEither.of(
+                telemetryClient.trackEvent({
+                  name: "api.messages.create.push",
```
Comment (francescopersico) [inline HandleNHNotificationCallActivity/handler.ts:112]:
```suggestion
                  name: "api.messages.notification.push",
```

File: HandleNHNotificationCallActivity/handler.ts [hunk 4]
```diff
@@ -98,9 +102,22 @@ export const getCallNHServiceActivityHandler = (
...
-            retryActivity(context, `${logPrefix}|ERROR=${toString(e)}`)
-          );
+          return notify(message.installationId, message.payload)
+            .mapLeft(e =>
+              retryActivity(context, `${logPrefix}|ERROR=${toString(e)}`)
+            )
+            .chainFirst(
+              taskEither.of(
+                telemetryClient.trackEvent({
+                  name: "api.messages.notification.push",
```
Comment (francescopersico) [inline HandleNHNotificationCallActivity/handler.ts:112]:
```suggestion
                  name: "api.messages.notification.push.sent",
```

We will send "api.messages.notification.push.queued" in io-functions-services

File: HandleNHNotificationCall/index.ts [hunk 5]
```diff
@@ -12,21 +13,38 @@ export const NotificationMessage = t.union([
...
+
+export type NhNotificationOrchestratorInput = t.TypeOf<
+  typeof NhNotificationOrchestratorInput
+>;
+
 // Initialize application insights
 initTelemetryClient();
 
 /**
  * Invoke Orchestrator to manage Notification Hub Service call with data provided by an enqued message
```
Comment (francescopersico) [inline HandleNHNotificationCall/index.ts:32]: We also need to change this

Comment (gunzip) [reply HandleNHNotificationCall/index.ts:32]: in which sense?

File: HandleNHNotificationCallActivity/handler.ts (hunk 2 above)
Comment (gunzip) [reply HandleNHNotificationCallActivity/handler.ts]: that's the latest step of https://github.com/pagopa/io-functions-services/pull/100

**Comment by balanza** [comment]
not needed anymore

**Pull request closed without merging**